rotkehlchen/exchanges/kucoin.py

```diff
@@ -0,0 +1,775 @@
+
+                # Trigger retry
+                log.debug(
+                    f'Kucoin {case} request reached the rate limits. Backing off',
+                    seconds=retries_after_seconds,
+                    options=call_options,
+                )
+                retries_left -= 1
+                retries_after_seconds *= 2
+                gevent.sleep(retries_after_seconds)
+                continue
+
+            if response.status_code != HTTPStatus.OK:
+                return self._process_unsuccessful_response(
+                    response=response,
+                    case=case_,
+                )
+
+            try:
+                response_dict = rlk_jsonloads_dict(response.text)
+            except JSONDecodeError as e:
+                msg = f'Kucoin {case} returned an invalid JSON response: {response.text}.'
+                log.error(msg)
+                self.msg_aggregator.add_error(
+                    f'Got remote error while querying kucoin {case}: {msg}',
+                )
+                raise RemoteError(msg) from e
+
+            try:
+                response_data = response_dict['data']
+                total_page = response_data['totalPage']
+                current_page = response_data['currentPage']
+                raw_results = response_data['items']
+            except KeyError as e:
+                msg = f'Kucoin {case} JSON response is missing key: {str(e)}'
+                log.error(msg, response_dict)
+                raise RemoteError(msg) from e
+
+            is_below_start_ts = False
+            for raw_result in raw_results:
+                try:
+                    result, skip_reason = deserialization_method(
+                        raw_result=raw_result,
+                        start_ts=start_ts,
+                        end_ts=end_ts,
+                    )
+                except DeserializationError as e:
+                    self.msg_aggregator.add_error(
+                        f'{str(e)}. Check logs for more details',
+                    )
+                    continue
+
+                if result is None and skip_reason is not None:
+                    if skip_reason == SkipReason.BELOW_TIMESTAMP_RANGE:
+                        log.debug(f'Found a kucoin {case} {str(skip_reason)}. Stop requesting.')
+                        break
+
+                    if skip_reason in (
+                        SkipReason.ABOVE_TIMESTAMP_RANGE,
+                        SkipReason.INNER_MOVEMENT,
+                    ):
+                        log.debug(f'Found a kucoin {case} {str(skip_reason)}. Skipping it.')
+                        continue
+
+                    raise AssertionError(f'Unexpected skip reason: {skip_reason}. Update conditions')  # noqa: E501
+
+                if result is None:
+                    # NB: may happen only during development
+                    raise AssertionError(f'Unexpected return deserializing {case} calling {endpoint}.')  # noqa: E501
```

Comment:
I'm doubtful about how to handle this case. It should never happen but during development (a mistake in a deserializer) and there is a bit of pleasing mypy (it moans as suddenly result can be None as well).